io/js/src/main/scala/fs2/io/file/FilesPlatform.scala

```diff
@@ -167,20 +168,40 @@ private[fs2] trait FilesCompanionPlatform {
           false
         }
 
+    override def getBasicFileAttributes(path: Path, followLinks: Boolean): F[BasicFileAttributes] =
+      getPosixFileAttributes(path, followLinks).widen
+
     override def getLastModifiedTime(path: Path, followLinks: Boolean): F[FiniteDuration] =
       stat(path, followLinks).map { stats =>
         stats.mtimeMs.milliseconds
       }
 
-    override def getPosixPermissions(path: Path, followLinks: Boolean): F[PosixPermissions] =
-      stat(path, followLinks).flatMap { stats =>
-        val value = stats.mode.toInt & 511
-        F.fromOption(
-          PosixPermissions.fromInt(value),
-          new IOException(s"Illegal PosixPermissions: ${value.toOctalString}")
-        )
+    def getPosixFileAttributes(path: Path, followLinks: Boolean): F[PosixFileAttributes] =
+      stat(path, followLinks).map { stats =>
+        new PosixFileAttributes.UnsealedPosixFileAttributes {
+          def creationTime: FiniteDuration = stats.ctimeMs.milliseconds
+          def fileKey: Option[FileKey] = if (stats.dev != 0 || stats.ino != 0)
+            Some(PosixFileKey(stats.dev.toLong, stats.ino.toLong))
+          else None
+          def isDirectory: Boolean = stats.isDirectory()
+          def isOther: Boolean = !isDirectory && !isRegularFile && !isSymbolicLink
+          def isRegularFile: Boolean = stats.isFile()
+          def isSymbolicLink: Boolean = stats.isSymbolicLink()
+          def lastAccessTime: FiniteDuration = stats.atimeMs.milliseconds
+          def lastModifiedTime: FiniteDuration = stats.mtimeMs.milliseconds
+          def size: Long = stats.size.toLong
+          def owner: Principal = ???
+          def group: Principal = ???
+          def permissions: PosixPermissions = {
+            val value = stats.mode.toInt & 511
```

Comment:
What is this `& 511`?

Reply:
A file mode contains two pieces of information, the upper bits are about the file type and the lower bits are the Posix permissions. This is a bit mask to retrieve these lower bits.